src/main/kotlin/com/github/mitchellolsthoorn/testgenie/coverage/TestGenieCoverageRenderer.kt

```diff
@@ -0,0 +1,84 @@
+package com.github.mitchellolsthoorn.testgenie.coverage
+
+import com.intellij.codeInsight.hint.HintManager
+import com.intellij.codeInsight.hint.HintManagerImpl
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.editor.LogicalPosition
+import com.intellij.openapi.editor.ex.EditorGutterComponentEx
+import com.intellij.openapi.editor.markup.ActiveGutterRenderer
+import com.intellij.openapi.editor.markup.LineMarkerRendererEx
+import com.intellij.ui.HintHint
+import com.intellij.ui.LightweightHint
+import com.intellij.ui.components.JBLabel
+import com.intellij.util.ui.FormBuilder
+import java.awt.Color
+import java.awt.Graphics
+import java.awt.Rectangle
+import java.awt.event.MouseEvent
+
+class TestGenieCoverageRenderer(private val color: Color, private val lineNumber : Int, private val tests : List<String>) : ActiveGutterRenderer,
+    LineMarkerRendererEx {
+
+    /**
+     * Perform the action - show toolTip on mouse click.
+     *
+     * @param editor the editor
+     * @param e mouse event
+     */
+    override fun doAction(editor: Editor, e: MouseEvent) {
+        e.consume()
```

Comment:
What does `consume` do?

Reply:
You can consume the event, so that it is not used by other methods later on. I read that it is good idea to consume events.

Reply:
Interesting